src/common/ceph_crypto.cc

```diff
@@ -83,6 +84,7 @@ void ceph::crypto::shutdown()
   assert(crypto_refs > 0);
   if (--crypto_refs == 0) {
     NSS_ShutdownContext(crypto_context);
+    PR_Cleanup();
```

Comment:
ahh, right. but no, it's not something should only be called by daemon and apps.


i looked at source of NSPR. the point of 

> may be called by the primordial thread near the end of the main() function

is that, the resource cleaned up by this function is **shared** globally by all NSS context. that's why the NSPR's comment put this way. but we refcounts the `crypto_context` users, this ensures us that it's safe to call this way.